src/rateLimiters/slidingWindowCounter.ts

```diff
@@ -0,0 +1,95 @@
+import Redis from 'ioredis';
+import { RateLimiter, RateLimiterResponse, RedisWindow } from '../@types/rateLimit';
+
+/**
+ * The SlidingWindowCounter instance of a RateLimiter limits requests based on a unique user ID.
+ * This algorithm improves upon the FixedWindowCounter because this algorithm prevents fixed window's
+ * flaw of allowing doubled capacity requests when hugging the window's borders with a rolling window,
+ * allowing us to average the requests between both windows proportionately with the rolling window's
+ * takeup in each.
+ *
+ * Whenever a user makes a request the following steps are performed:
+ *  1. Fixed minute windows are defined along with redis caches if previously undefined.
+ *  2. Rolling minute windows are defined or updated based on the timestamp of the new request.
+ *  3. Counter of the current fixed window is updated with the new request's token usage.
+ *  4. If a new minute interval is reached, the averaging formula is run to prevent fixed window's flaw
+ *     of flooded requests around window borders
+ *    (ex. 10 token capacity: 1m59s 10 reqs 2m2s 10 reqs)
+ */
+class SlidingWindowCounter implements RateLimiter {
+    private windowSize: number;
+
+    private capacity: number;
+
+    private client: Redis;
+
+    /**
+     * Create a new instance of a TokenBucket rate limiter that can be connected to any database store
```

Comment:
This should say `SlidingWindowCounter` instead of `TokenBucket`

Reply:
done